pages/docs/internals/how-it-works.md

```diff
@@ -0,0 +1,299 @@
+
+Authors of CSR websites/apps should be aware that rendering will happen after the HTTP response for the site configuration completes.
+This means that the `InstantBandit` component and its children will "pop in" asynchronously, and this should be considered when placing the component.
+
+See [Performance Tips](../tips.md#performance) for information about how to optimie performance in CSR.
+
+
+## Context
+The Instant Bandit component provides a React Context: the `InstantBanditContext`.
+
+Any children of an `InstantBandit` component have access to this context.
+The context bears information about all of the experiments, variants, and metadata for a site.
+
+Components can use the `useInstantBandit` hook to gain imperative API access.
+Most commonly, this is done to record `conversion` events.
+
+The use of context means that components remain reusable across variants. Consider this page and component:
+
+```tsx
+// index.tsx
+<InstantBandit>
+  <Default>
+    <h2>... existing CTA text ...</h2>
+  </Default>
+
+  <Variant name="call-to-action-text-new">
+    <h2>... new CTA text ...</h2>
+  </Variant>
+
+  <SignUpButton>Add Conversion</SignUpButton>
+</InstantBandit>
+
+// inside SignUpButton.tsx
+const { incrementMetric } = useInstantBandit();
+
+const onClick = useCallback(() => {
+  incrementMetric(DefaultMetrics.CONVERSIONS);
+}, [ metrics]);
+
+```
+Here, we're testing some existing CTA text against a new variant.
+
+The `SignUpButton` in this example doesn't need to know anything about the current variant.
+The call to `incrementMetric` via `metrics` API is bound to the context; the metric will be recorded in the correct block of metrics for the current variant.
+
+This keeps things simple, as components that record metrics need not concern themselves with specific variants and are reusable.
+
+In the case of no active experiment, or some error preventing the site config from loading, the `exposures` and `conversions` metrics in our example will simply be tracked against the invariant.
+
+See [Working with Metrics](./working-with-metrics.md) for information on capturing and analyzingmetrics.
+
+
+## Returning Visitors
+Returning visitors who haved previously been exposed to a variant will see the same variant they did
+before, and any further `exposures` or `conversions` from them will be tracked against their current variant.
+
+
+## Experiment Lifecycle
+Experiments run for as long as they are considered active in the site config.
+
+When you feel that you've obtained statistical significance based on the observed metrics for the variants in a given experiment, you can mark the current experiment inactive, and promote the winning variant into permanent status by removing the `Variant` from around it.
+
+You should then mark the experiment as inactive like so:
+
+```json
+```
+
+You can leave it in the site configuration as a historical
```

Comment:
Seems to trail off, should say something along the lines of "to maintain a history"?